src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Ink/LassoHelper.cs

```diff
@@ -185,9 +184,9 @@ public bool ArePointsInLasso(Point[] points, int percentIntersect)
             // If the no of such segments is odd then the point is within the lasso otherwise not.
             int countPointsInLasso = 0;
 
-            foreach (Point point in points)
+            foreach (ref Point point in points)
```

Comment:
Could show the performance improvement of only adding these keywords here because I ran a reduced benchmark for this and I didn't see any improvements.

Reply:
`Point` is a very small structure. Parameters may be passed using registers.

Reply:
@ThomasGoulet73 the assembly JIT generates is very slightly different, you trade off copy of 16 bytes for a mem ref, which is when it is actually worth it already on x64 if it can save you a copy on stack. It's about 2ns (27ns vs 25ns via ref) from using it like this but as I wrote in the original description, I have no issues with changing it because I see the other side :)

Reply:
@h3xds1nz Thank you. But I'm confused about the data from your test results. Because I do not think it costs 2ns, 2ns is too long.

Reply:
@lindexi I can't get a proper benchmark that's conclusive, but it's always between 1 and 2ns diff for 20/20 in favour of ref. I know from the codegen it spills and it saves non-volatile registers as it uses them as for copying in case of copies. Eitherway I'll just do the regular code, lemme push the commit with ROS. No point wasting time on that, it's just my years of coding in assembly.